giraffe/src/types/index.ts

```diff
@@ -689,17 +689,18 @@ export interface RectLayerSpec {
 }
 
 export interface LayerProps {
-  xScale: Scale<number, number>
-  yScale: Scale<number, number>
-  width: number
-  height: number
-  spec: LayerSpec
+  columnFormatter: (colKey: string) => (x: any) => string
+  yColumnType: ColumnType
   config: LayerConfig
   plotConfig: SizedConfig
-  yColumnType: ColumnType
-  columnFormatter: (colKey: string) => (x: any) => string
+  spec: LayerSpec
   hoverX: number | null
   hoverY: number | null
+  legendHide: boolean
```

Comment:
❓ :  Since this is not an optional parameter, would it be a breaking change in ui?

Reply:
It is not optional but will not break the ui because this is in `LayerProps` which is only an internal type. It does get exported out of index.ts but it has not been exported at the library level.